kore/src/Kore/Equation/Application.hs

```diff
@@ -156,6 +157,7 @@ attemptEquation
     -> Equation RewritingVariableName
     -> simplifier (AttemptEquationResult RewritingVariableName)
 attemptEquation sideCondition termLike equation =
+    assert (not $ hasFreeEquationVariableName termLike) $
```

Comment:
Do we ever use `hasFreeEquationVariableName` on its own (without negating it)? Could we just turn `hasFreeEquationVariableName` into `withoutFreeEquationVariables`?